### DISCUSSION TITLE
Upgrade to 1.1.0.RC10 - Where to Start?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#1067] [General] [answered]
Hi,

I realized that my current code base of my diagram project is still based on `1.1.0-RC04` and so I planed to upgrade to latest RC release `1.1.0.RC10`.

I expected some changes, but it seems this is a massive API change :-( I looks like I have API breaks in all code areas. 

My question is: Where can I found example code based on RC10 ?

**Comment by rsoika** [comment]
I also fail to checkout the glsp server project RC10 in VSCode or Eclipse. I get to many dependency issues and I can't even analyse the `org.eclipse.glsp.example.workflow` . 

For example in `org.eclipse.glsp.server.gmodel.GModelCreateNodeOperationHandler`  the method `getLocation` can't be found and can not be overwritten


```java
   @Override
   public Optional<GPoint> getLocation(final CreateNodeOperation operation) {
      return GridSnapper.snap(operation.getLocation());
   }
```
In my own project I use the following maven dependencies:


		<dependency>
			<groupId>com.google.guava</groupId>
			<artifactId>guava</artifactId>
			<version>[30.1-jre,)</version>
		</dependency>
		<dependency>
			<groupId>org.eclipse.glsp</groupId>
			<artifactId>org.eclipse.glsp.server</artifactId>
			<version>1.1.0.RC10</version>
		</dependency>
		<dependency>
			<groupId>org.eclipse.glsp</groupId>
			<artifactId>org.eclipse.glsp.graph</artifactId>
			<version>1.1.0.RC10</version>
		</dependency>	
		<dependency>
			<groupId>org.eclipse.glsp</groupId>
			<artifactId>org.eclipse.glsp.server.websocket</artifactId>
			<version>1.1.0.RC10</version>
		</dependency>


Is this correct?

**Comment by rsoika** [comment]
So far I found out that the `org.eclipse.glsp.server.operations.AbstractCreateOperationHandler` was replaced by `org.eclipse.glsp.server.gmodel.GModelCreateOperationHandler`


So I change my old creation pattern from 

```java 
    @Override
    protected void executeOperation(final CreateNodeOperation operation) {
        elementTypeId = operation.getElementTypeId();

        try {
           ....
.....        } catch (BPMNModelException e) {
            e.printStackTrace();
        }
        modelState.reset();
        actionDispatcher.dispatchAfterNextUpdate(new SelectAction(List.of(dataObjectID)));
    }
```

into this:

```java 
    @Override
    public Optional<Command> createCommand(final CreateNodeOperation operation) {
        return commandOf(() -> executeOperation(operation));
    }

    protected void executeOperation(final CreateNodeOperation operation) {
        elementTypeId = operation.getElementTypeId();

        try {
           ....
.....        } catch (BPMNModelException e) {
            e.printStackTrace();
        }
        modelState.reset();
        actionDispatcher.dispatchAfterNextUpdate(new SelectAction(List.of(dataObjectID)));
    }
```

Is this the correct way to create a node?

**Comment by rsoika** [comment]
Hm... anyway I still failed to build my Server project with GLSP 1.1.0-RC10.

At the end I got the following maven error message:

` Failure to find org.eclipse.glsp:org.eclipse.glsp.graph:eclipse-plugin:1.1.0.RC10 in https://oss.sonatype.org/content/repositories/snapshots was cached in the local repository,`

Currently I have no idea how to resolve this

**Reply by rsoika** [reply]
Ok, the solution for the missing maven artefacts seems to be the following procedure

1. checkout the glsp-server project tag 1.1.0-RC10
2. build the project locally with
    `$ mvn clean install -Pm2`

Now you have the necessary maven artefacts in you local repo. 

And in my final pom.xml I only add the glsp dependency

```
<dependency>
	<groupId>org.eclipse.glsp</groupId>
	<artifactId>org.eclipse.glsp.server</artifactId>
	<version>1.1.0.RC10</version>
</dependency>
```

But I think building the glsp-server locally should not be necessary. So this is still a strange situation.

**Comment by rsoika** [comment, marked as answer]
Hooray! I did it! 
Finally the application builds and I can start the Theia app locally. 

Ok, you can't open diagrams any more, but I hope that this can be debugged now.....

So the important steps so far for upgrade from `1.1.0-RC04` to `1.1.0-RC10` are:

1) changes from the `AbstractCreateOperationHandler` to the new Java class `GModelCreateOperationHandler`  
2) the need to build the glsp-server module locally - don't know why this was necessary in my case
3) some smaller changes in the client API regarding `createDiagramContainer` and `initializeDiagramContainer`

**Reply by tortmayr** [reply]
Good to hear.
Unfortunatley at the moment no migration guide for migrating from 1.00 to the newest RC release is available. We plan to prepare a guide for the 2.0 Release at the end of August.

If you need a reference and what has changed I recommend to have a look at the changes in the workflow-dev example of the corresponding repos. You could even compare the 1.1.0-RC04 branch against the 1.1.0-RC10 branch to see what has changed.

**Reply by rsoika** [reply]
@tortmayr, yes I have done so. But it's hard to get the background for changes only out of the code.
I am hoping that Release 2.0 is not again a hard API switch?

**Comment by rsoika** [comment]
Hi @tortmayr ,  just to give you more feedback: 

I seems that I have now fixed all issues regarding the upgrade from RC04 -> RC10 . 
There were some more layout issues. Indeed I have had some strange and inconsistent code with RC04 . Now, after all the fixes everything looks much better and more consistent regarding the layout of elements.
